pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -164,18 +170,24 @@ func IsUpgradeServiceAffecting(csv *olmv1alpha1.ClusterServiceVersion) bool {
 }
 
 func ApproveUpgrade(ctx context.Context, client k8sclient.Client, installPlan *olmv1alpha1.InstallPlan, eventRecorder record.EventRecorder) error {
+	if installPlan.Spec.Approved == true {
+		return nil
+	}
```

Comment:
This should not be needed. The check below is also checking if the installplan install is in progress.

```	
if installPlan.Status.Phase == olmv1alpha1.InstallPlanPhaseInstalling {
	return nil			
}
```

Reply:
I found if I removed this check that I was getting quite a lot of noise in the logs in the form of: "the object has been updated", adding this check kept everything functional but stopped that noise.

Reply:
That makes sense, thanks @philbrookes